internal/config/attestation.go

```diff
@@ -13,9 +13,50 @@ import (
 	"fmt"
 
 	"github.com/edgelesssys/constellation/v2/internal/attestation/measurements"
+	"github.com/edgelesssys/constellation/v2/internal/cloud/cloudprovider"
 	"github.com/edgelesssys/constellation/v2/internal/variant"
 )
 
+type AttestationType (string)
+
+const (
+	AttestationTypeAWSNitroTPM        AttestationType = "aws-nitro-tpm"
+	AttestationTypeAzureSEVSNP        AttestationType = "azure-sev-snp"
+	AttestationTypeAzureTrustedLaunch AttestationType = "azure-trusted-launch"
+	AttestationTypeGCPSEVES           AttestationType = "gcp-sev-es"
+	AttestationTypeQEMUVTPM           AttestationType = "qemu-vtpm"
```

Comment:
```suggestion
	AWSNitroTPM        AttestationType = "aws-nitro-tpm"
	AzureSEVSNP        AttestationType = "azure-sev-snp"
	AzureTrustedLaunch AttestationType = "azure-trusted-launch"
	GCPSEVES           AttestationType = "gcp-sev-es"
	QEMUVTPM           AttestationType = "qemu-vtpm"
```
Id say the information is already encoded in the type name.

Reply:
But that doesn't help when reading code on the caller side. As the package name is `config`, it would read `config.QEMUVTPM` , which isn't enough information Imo. Would totally agree if this had its own package `attestation` .

Reply:
We already have [`variant`](https://github.com/edgelesssys/constellation/blob/main/internal/variant/variant.go), could just export the string constants, if working with strings is required.

Reply:
`variant` is a really undescriptive package name. :laughing:

Reply:
I looked at the places where the variables are used ([1](https://github.com/edgelesssys/constellation/pull/1769/files#diff-54c7c1af5fa8d5db4dc49f0e8e80e93ba2b1183ba4d5c9e2e5729e6deae6a3cdR468-R478), [2](https://github.com/edgelesssys/constellation/pull/1769/files#diff-438814676158a6febbb3ef8e77769ba76f6d70d7df4bac31c3101b1e0b64c3b4R185), [3](https://github.com/edgelesssys/constellation/pull/1769/files#diff-62a8c60cf08995632a7054539677e3f3b04a0bfb956018e12ac6804ec4a9aebbR147-R153), ...) and felt like the type information was always close to the use of the const and thus it would be pretty readable. But feel free to skip this.

Reply:
I skipped the renaming due to the conflict with another equally named struct. They could be merged but I don't see the benefit at the moment to invest more refactoring.